src/Connections/GitlabRepo.ts

```diff
@@ -291,7 +295,39 @@ export class GitLabRepoConnection extends CommandConnection<GitLabRepoConnection
         }
     }
 
-    public static async getConnectionTargets(userId: string, tokenStore: UserTokenStore, config: BridgeConfigGitLab, filters: GitLabTargetFilter = {}): Promise<GitLabRepoConnectionTarget[]> {
+    public static async getBase64Avatar(avatarUrl: string, client: GitLabClient, storage: IBridgeStorageProvider): Promise<string|null> {
+        try {
+            const existingFile = await storage.getStoredTempFile(avatarUrl);
+            if (existingFile) {
+                return existingFile;
+            }
+            const res = await client.get(avatarUrl);
+            if (res.status !== 200) {
+                return null;
+            }
+            const contentType = res.headers["content-type"];
+            if (!contentType?.startsWith("image/")) {
+                return null;
+            }
+            const data = res.data as Buffer;
+            const url = `data:${contentType};base64,${data.toString('base64')}`;
+            await storage.setStoredTempFile(avatarUrl, url);
+            return url;
+        } catch (ex) {
+            if (axios.isAxiosError(ex)) {
+                if (ex.response?.status === 401) {
+                    // 401 means that the project is Private and GitLab haven't fixed
+                    // the auth issues, just ignore this one.
+                    // https://gitlab.com/gitlab-org/gitlab/-/issues/25498
+                    return null;
+                }
+            }
+            log.warn(`Could not transform data from ${avatarUrl} into base64`, ex);
```

Comment:
You only tested that it's not an HTTP 404 error.
This could still be an HTTP 500 or other run time error.

Also, I don't think there is any Buffer which cannot be transformed to base64.

Reply:
It's okay for this function to fail for any reason, we just want to log if the error was unexpected. However, let's improve the error reporting.